### PR TITLE
[nginx] allow configuring maximum client certificate chain length

### PR DESCRIPTION
**Pull request by mikz**
some clients need to send the whole chain (certificate, intermediate, root),
and nginx needs to accept that whole chain to pass it to policies

Also fixes a segfault when parsing some certificates (intermediate CA, EC). Can be reproduced with the certificates committed in the repo.

# TODO

- [x] integration test using the full chain

### CODE REVIEW

File: doc/parameters.md [hunk 1]
```diff
@@ -155,6 +155,14 @@ Path to a file with passphrases for the SSL cert keys specified with
 - `on`: reuses SSL sessions.
 - `off`: does not reuse SSL sessions.
 
+### `APICAST_PROXY_HTTPS_VERIFY_DEPTH`
+
+**Default:** 1
+**Values:** positive integers
+
+Defines the maximum length of the client certificate chain.
+Length 1 means there can be 1 extra certificate included (eg. intermediate CA).
```
Comment (porueesq) [inline doc/parameters.md]: Suggestion:
_If this parameter has `1` as its value, it implies that this length might include one additional certificate  (eg. intermediate CA)._

Comment (mikz) [reply doc/parameters.md]: 👍

File: gateway/src/resty/openssl/x509.lua [hunk 2]
```diff
@@ -55,7 +55,8 @@ local function normalize_pem_cert(str)
   if not str then return end
   if #(str) == 0 then return end
 
-  return re_gsub(str, [[\s(?!CERTIFICATE)]], '\n', 'oj')
+  -- using also jit compiler (j) will result in a segfault with some certificates
```
Comment (davidor) [inline gateway/src/resty/openssl/x509.lua]: Do you know which certificates cause the problem?

File: t/fixtures/CA/Makefile [hunk 3]
```diff
@@ -0,0 +1,29 @@
+all: client.key client.crt server.key server.crt ca-bundle.crt:
```
Comment (davidor) [inline t/fixtures/CA/Makefile]: This Makefile is going to be super useful :+1:

File: gateway/src/resty/openssl/x509.lua (hunk 2 above)
Comment (mikz) [reply gateway/src/resty/openssl/x509.lua]: The updated ones in this PR: t/fixtures/CA.  But only when they come from nginx, couldn't reproduce it in busted.

File: t/fixtures/CA/Makefile (hunk 3 above)
Comment (mikz) [reply t/fixtures/CA/Makefile]: Yeah, got tired of figuring out how to generate them every time 😄

File: doc/parameters.md [hunk 4]
```diff
@@ -155,6 +155,14 @@ Path to a file with passphrases for the SSL cert keys specified with
 - `on`: reuses SSL sessions.
 - `off`: does not reuse SSL sessions.
 
+### `APICAST_PROXY_HTTPS_VERIFY_DEPTH`
```
Comment (mikz) [inline doc/parameters.md]: This is wrong. Fixed it in https://github.com/3scale/APIcast/pull/1048

**Pull request merged**